### PR TITLE
Add CI data from Green Algorithms Project

### PR DESCRIPTION
**Pull request by skrakau**
Copied over location-specific CI data from the Green Algorithms project, adjusted handling and added (very brief) description to docs.

### CODE REVIEW

**Review by mirpedrol — APPROVED** [review]
LGTM

File: docs/usage/parameters.md [hunk 1]
```diff
@@ -27,5 +27,9 @@ The following parameters are currently available:
 - `file`: Name of the TXT carbon footprint report containing the energy consumption, the estimated CO<sub>2</sub> emission and other relevant metrics for each task.
 - `reportFile`: Name of the HTML report containing information about the entire carbon footprint, overview plots and more detailed task-specific metrics.
 - `ci`: carbon intensity of the respective energy production.
+- `location`: location code to automatically retrieve a location-specific CI value.
+For countries, these are [ISO alpha-2 codes](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2). 
+For regions, it’s the ISO alpha-2 code for the country, followed by an identifier for the state, e.g. US-CA for “California, USA”.
+You can find the available data [here](../../plugins/nf-co2footprint/src/resources/CI_aggregated.v2.2.csv).
```
Comment (Llannelongue) [inline docs/usage/parameters.md:33]: Would it be worth clarifying here that either `ci` or `location` should be used, and if both are specified, which one takes over? (It should probably be `ci`). (Sorry if this is done elsewhere!)
e.g. :
- adding "Only used if CI is not specified" to the description of `location` 
- and adding to the description of `ci ` above: "If not specified, `location` should be used"

Comment (Llannelongue) [reply docs/usage/parameters.md:33]: Ah I see below that having both raises an error!

File: plugins/nf-co2footprint/src/main/nextflow/co2footprint/CO2FootprintConfig.groovy [hunk 2]
```diff
@@ -59,16 +59,16 @@ class CO2FootprintConfig {
         file = config.file ?: CO2FootprintFactory.CO2FootprintTextFileObserver.DEF_FILE_NAME
         summaryFile = config.summaryFile ?: CO2FootprintFactory.CO2FootprintTextFileObserver.DEF_SUMMARY_FILE_NAME
         reportFile = config.reportFile ?: CO2FootprintFactory.CO2FootprintReportObserver.DEF_REPORT_FILE_NAME
-        def country
 
         ci = 475
         if (config.ci && config.country)
-            throw new IllegalArgumentException("Invalid combination of 'ci' and 'country' parameters specified for the CO2Footprint plugin. Please specify either 'ci' and 'country'!")
+            throw new IllegalArgumentException("Invalid combination of 'ci' and 'location' parameters specified for the CO2Footprint plugin. Please specify either 'ci' and 'location'!")
```
Comment (Llannelongue) [inline plugins/nf-co2footprint/src/main/nextflow/co2footprint/CO2FootprintConfig.groovy:65]: Shouldn't the last part read "please specify either 'ci' OR 'location'"?

**Review by Llannelongue — APPROVED** [review]

File: plugins/nf-co2footprint/src/main/nextflow/co2footprint/CO2FootprintConfig.groovy (hunk 2 above)
Comment (skrakau) [reply plugins/nf-co2footprint/src/main/nextflow/co2footprint/CO2FootprintConfig.groovy:65]: Ups, yes of course

File: docs/usage/parameters.md (hunk 1 above)
Comment (skrakau) [reply docs/usage/parameters.md:33]: I added a note that they are mutually exclusive now

**Pull request merged**